src/PointwiseFunctions/MathFunctions/MathFunction.hpp

```diff
@@ -0,0 +1,90 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+/// \file
+/// Defines base-class MathFunction.
+
+#pragma once
+
+#include <memory>
+
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Options/Factory.hpp"
+
+/// Holds classes implementing MathFunction (functions \f$R\to R\f$).
+namespace MathFunctions {
+class Gaussian;
+class PowX;
+class Sinusoid;
+}  // namespace MathFunctions
+
+/*!
+ *  \ingroup MathFunctions
+ *  Encodes a function \f$R^n\to R\f$ where n is `VolumeDim`.
+ */
+template <size_t VolumeDim>
+class MathFunction : public Factory<MathFunction<VolumeDim>> {
+ public:
+  using creatable_classes = typelist<>;
+
+  MathFunction() = default;
+  MathFunction(const MathFunction& /*rhs*/) = delete;
+  MathFunction& operator=(const MathFunction& /*rhs*/) = delete;
+  MathFunction(MathFunction&& /*rhs*/) noexcept = default;             // NOLINT
+  MathFunction& operator=(MathFunction&& /*rhs*/) noexcept = default;  // NOLINT
+  virtual ~MathFunction() = default;
+
+  /// Returns the function value at the coordinate 'x'.
+  template <typename T, typename Fr>
+  T operator()(const tnsr::I<T, VolumeDim, Fr>& x) const;
+
+  /// Returns the first derivative at 'x' as a one-form.
+  template <typename T, typename Fr>
+  tnsr::i<T, VolumeDim, Fr> first_deriv(
+      const tnsr::i<T, VolumeDim, Fr>& x) const;
+
+  /// Returns the matrix of second derivatives at 'x'.
+  template <typename T, typename Fr>
+  tnsr::ii<T, VolumeDim, Fr> second_deriv(T x) const;
+};
+
+/*!
+ * \ingroup MathFunctions
+ * Partial template specialization of MathFunction which encodes a
+ * function \f$R \to R\f$.
+ */
+template <>
+class MathFunction<1> : public Factory<MathFunction<1>> {
+ public:
+  using creatable_classes = typelist<
+      // MathFunctions::Gaussian,
+      MathFunctions::PowX
+      // ,MathFunctions::Sinusoid
+      >;
+
+  MathFunction() = default;
+  MathFunction(const MathFunction& /*rhs*/) = delete;
+  MathFunction& operator=(const MathFunction& /*rhs*/) = delete;
+  MathFunction(MathFunction&& /*rhs*/) noexcept = default;             // NOLINT
+  MathFunction& operator=(MathFunction&& /*rhs*/) noexcept = default;  // NOLINT
+  virtual ~MathFunction() = default;
+
+  /// Returns the function value at the coordinate 'x'.
+  template <typename T>
+  T operator()(const T& x) const;
```

Comment:
These functions need to be virtual or similar.  Probably the simplest way to do that here is to overload for double and DataVector, although it will be a bit annoying in the derived classes.

Also, do we want versions of these matching the ones in the general base class (that are presumably just simple forwards)?  It would make generic stuff easier, but I'm not sure how much that matters here.  I guess generally how closely are we going to connect the general class and the 1D specialization?